pkg/process/config/config.go

```diff
@@ -520,6 +520,36 @@ func constructProxy(host, scheme string, port int, user, password string) (proxy
 	return http.ProxyURL(u), nil
 }
 
+// getFargateHost returns the hostname to be used
+// in the config based on the Fargate orchestrator
+// - ECS: fargate_task:<TaskARN>
+// - EKS: value of kubernetes_kubelet_nodename
+func getFargateHost(orchestrator fargate.OrchestratorName, ecsFunc, eksFunc func() (string, error)) (string, error) {
+	// Fargate should have no concept of host names
+	// we set the hostname depending on the orchestrator
+	switch orchestrator {
+	case fargate.ECS:
+		return ecsFunc()
+	case fargate.EKS:
+		return eksFunc()
+	}
+	return "", errors.New("unknown Fargate orchestrator")
+}
+
+func getECSHost() (string, error) {
+	// Use the task ARN as hostname
+	taskMeta, err := ecsmeta.V2().GetTask()
+	if err != nil {
+		return "", err
+	}
+	return fmt.Sprintf("fargate_task:%s", taskMeta.TaskARN), nil
+}
+
+func getEKSHost() (string, error) {
+	// Use the node name as hostname
+	return fargate.GetEKSFargateNodename()
+}
+
```

Comment:
I think this should go into the fargate package (the config package should not know the difference between each orchastrator).

Reply:
Thanks! From a design perspective I totally agree, but we were keeping this here (as close as possible to the process agent) because the process backend handles data coming from Fargate separately, so any changes here in the agent can break the logic there. 
Also we're handling Fargate not in the same way in the core agent (we set an empty hostname).
I'm thinking maybe we can move this logic to the fargate pkg and expose a function named `GetFargateHostForProcessAgent` ? like this we reduce the risk of breaking things in the future. wdyt?

Reply:
So, as a general note, the API of each package of the agent is not guarantee to be backward compatible. We set that expectation from the beginning for both out own backend and for users. We should pin the code pulled in the backend if we want to use it pieces of the agent. We need to be able to  change the code in the agent when needed (we don't wan to be limited by/tied to the backend).

For the solution I think a compile flag would be better. A `GetFargateHost` function compiled in the core agent and another on compiled in the process agent. We already have that setup in a majority of the code. See here for the compile flag list: https://github.com/DataDog/datadog-agent/blob/master/tasks/build_tags.py#L8

Reply:
thanks for the context 👍 
I updated the PR, let me know if you think of a better name for the new build tag `fargateprocess`